docs/api.md

```diff
@@ -0,0 +1,253 @@
+# soundslike - API documentation
+Learn how to use the soundslike API!
+
+First. for the sake of this, im going to use soundslike.com as the url, however localhost:8888 might be the url if you're using docker 
+
+## Section 1. Authentication 
+
+### Getting auth token from api
+
+run 'GET' http://soundslike.com/api/token with a username and password with basic AUTH  
+
+this will return a response of...
+```json
+    {
+        "expiration": 3000,
+        "token": "token_string"
+    }
+``` 
+you'll need to Base64 encode the bearer token provided earlier and include it in all API requests, except user creation 
+
+## Section 2. Users 
+
+### creating a new user 
+
+run 'POST' http://soundslike.com/api/users/
+
+with a application/json body of...
+```json
+{
+    "username" : "test", 
+    "email" : "test@gmail.com", 
+    "password": "test"
+}
+``` 
+and this will return a response of... 
+```json
+{
+    "email": "test@gmail.com",
+    "member_since": "Fri, 07 Dec 2018 06:19:28 GMT",
+    "username": "test"
+}
+``` 
+### getting a user 
+
+run 'GET' http://soundslike.com/api/users/method_man with no body and this will return a response of...
+```json
+{
+    "email": "method_man@wutang.com",
+    "member_since": "Fri, 07 Dec 2018 06:19:28 GMT",
+    "username": "method_man"
+}
+``` 
+NOTE: the email field is not guranteed unless you're accessing your own profile
+
+## Section 3. Songs 
+
+### getting a song
+
+run 'GET' http://soundslike.com/api/songs/2 with no body and this will return a response of...
+```json
+{
+  "artist": "Wartime Afternoon",
+  "created": "Thu, 06 Dec 2018 04:18:33 GMT",
+  "created_by": {
+    "email": "wartimeafternoon@gmail.com",
+    "member_since": "Tue, 04 Dec 2018 05:06:48 GMT",
+    "username": "wartimeafternoon"
+  },
+  "id": 2,
+  "title": "Marjik Janson",
+  "url": "https://www.youtube.com/watch?v=SUpV91yo1-M"
+}
+``` 
+
+### posting a song
+
+run 'POST' http://soundslike.com/api/songs/ with a body of 
```

Comment:
Can we use backticks instead of single quotes for that sweet sweet code markdown rendering?